block/bench/Block.hs

```diff
@@ -0,0 +1,249 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import           Universum
+
+import           Control.DeepSeq (NFData (..), deepseq)
+import           Criterion
+import           Criterion.Main
+import           Data.ByteString (ByteString)
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
+import           Formatting (sformat, build, shown)
+import           System.Environment (getEnv)
+
+import           Pos.Binary.Class (Bi, serialize, unsafeDeserialize)
+import           Pos.Block.Util.Generate
+import           Pos.Core.Common (SharedSeed (..), CoinPortion)
+import           Pos.Core.Configuration
+import           Pos.Core.Genesis
+import           Pos.Core (Block, BlockHeader, MainBlock, MainBlockHeader,
+                           BlockVersionData (..), SoftforkRule (..),
+                           EpochIndex (..), CoinPortion (..), unsafeCoinPortionFromDouble,
+                           Timestamp (..), TxFeePolicy (..), TxPayload (..),
+                           SscPayload, DlgPayload, UpdatePayload,
+                           MainBlockchain, BodyProof, ConsensusData,
+                           ExtraHeaderData, ExtraBodyData, Body,
+                           _gbBody, _gbHeader, _gbExtra, _mbTxPayload,
+                           _mbSscPayload, _mbDlgPayload, _mbUpdatePayload,
+                           _gbhBodyProof, _gbhConsensus, _gbhExtra)
+import           Pos.Core.Block.Main ()
+import           Pos.Crypto (ProtocolMagic (..))
+
+-- We need configurations in order to get Arbitrary and Bi instances for
+-- Block stuff.
+
+cc :: CoreConfiguration
```

Comment:
You could take compile-time configuration (`dev` or `devnet`) and change some parameters as you wish.